INSTALLING.rst

```diff
@@ -172,34 +221,56 @@ Add the ``--validate`` option to include longer-running validation tests:
 
    $ python3 -m pytest --pyargs gsd -p gsd.pytest_plugin_validate --validate
 
-Build user documentation
-^^^^^^^^^^^^^^^^^^^^^^^^
+.. tip::
 
-Build the user documentation with **Sphinx**. **IPython** is required to build the documentation, as is an active
-internet connection. First, you need to compile and install **gsd**. If you compiled with **CMake**, add **gsd**
-to your ``PYTHONPATH`` first:
+    When using CMake builds, change to the build directory before running ``pytest``::
 
-.. code-block:: bash
+        $ cd build/gsd
+
+.. _pytest: https://docs.pytest.org/
+
+.. _Build the documentation:
 
-   $ export PYTHONPATH=$PYTHONPATH:/path/to/build
+Build the documentation
+^^^^^^^^^^^^^^^^^^^^^^^
 
-To build the documentation:
+Run `Sphinx`_ to build the documentation:
 
 .. code-block:: bash
 
-   $ cd /path/to/gsd
-   $ cd doc
-   $ make html
-   $ open _build/html/index.html
+   $ sphinx-build -b html gsd/doc build/gsd-documentation
+
+Open the file :file:`build/gsd-documentation/index.html` in your web browser to view the
+documentation.
+
+.. tip::
+
+    When iteratively modifying the documentation, the sphinx options ``-a -n -W -T --keep-going``
+    are helpful to produce docs with consistent links in the side panel and to see more useful error
+    messages::
+
+        $ sphinx-build -a -n -W -T --keep-going -b html gsd/doc build/gsd-documentation
+
+.. tip::
+
+    When using CMake builds, change to the build directory before running ``sphinx-build``::
+
+        $ cd build/gsd
+        $ sphinx-build -b html ../../gsd/doc ../gsd-documentation
+
+.. _Sphinx: https://www.sphinx-doc.org/
```

Comment:
It's nice to include these extra tips. I'm no sphinx expert so these were enlightening.

Reply:
I'm moving away from the makefile that sphinx-quickstart generated and instead giving users the freedom to manually run sphinx-build (which many were doing anyways). However, the default options in sphinx leave a lot to be desired. If you have an import error, for example, sphinx will generate the docs and not display any errors or information to help you find the problem. It will happily produce documentation that is empty wherever autodoc failed to read the docstrings. These options help fix that. Also the `-a` option forces a rebuild of all files which is necessary when using the readthedocs sphinx theme. Otherwise partial rebuilds will result in inconsistent links in navigation pane.